packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -992,43 +995,147 @@ export class CodeModelBuilder {
+              sdkMethod.pagingMetadata.continuationTokenResponseSegments,
+            );
+            if (continuationTokenParameterSegment) {
+              // for now, continuationToken is either request query or header parameter
+              const parameter = getHttpOperationParameter(
+                sdkMethod,
+                continuationTokenParameterSegment,
+              );
+              if (parameter && op.parameters) {
+                for (const param of op.parameters) {
+                  if (param.protocol.http?.in === parameter.kind) {
+                    if (
+                      parameter.kind === "header" &&
+                      param.language.default.serializedName.toLowerCase() ===
+                        parameter.serializedName.toLowerCase()
+                    ) {
+                      continuationTokenParameter = param;
+                      break;
+                    } else if (param.language.default.serializedName === parameter.serializedName) {
+                      continuationTokenParameter = param;
+                      break;
+                    }
+                  }
+                }
+              }
             }
-          });
+            if (continuationTokenResponseSegment && op.responses) {
+              if (continuationTokenResponseSegment?.kind === "responseheader") {
+                // continuationToken is response header
+                for (const response of op.responses) {
+                  if (response instanceof SchemaResponse && response.protocol.http) {
+                    for (const header of response.protocol.http.headers) {
+                      if (
+                        header.header.toLowerCase() ===
+                        continuationTokenResponseSegment.serializedName.toLowerCase()
+                      ) {
+                        continuationTokenResponseHeader = header;
+                        break;
+                      }
+                    }
+                  }
+                  if (continuationTokenResponseHeader) {
+                    break;
+                  }
+                }
+              } else if (continuationTokenResponseSegment?.kind === "property") {
+                // continuationToken is response body property
+                // TODO: the property could be nested
+                for (const response of op.responses) {
+                  if (
+                    response instanceof SchemaResponse &&
+                    response.schema instanceof ObjectSchema &&
+                    response.schema.properties
+                  ) {
+                    for (const property of response.schema.properties) {
+                      if (
+                        property.serializedName ===
+                        getPropertySerializedName(continuationTokenResponseSegment)
+                      ) {
+                        continuationTokenResponseProperty = [property];
```

Comment:
curious; why `continuationTokenResponseProperty ` is an array?

Reply:
The design is to allow it be in nested property, e.g.
```
{
  "links": {
    "continuationToken": "###"
  }
}
```

So here we need a path to the property.

It actually should be same for "nextLink".

But currently the dependent lib cannot handle such case, hence the code here is a simple one right now (with a TODO a few lines above).

Reply:
I see, we use array to future proof, thanks for the clarification!